tests/cxx/badinc.cc

```diff
@@ -307,27 +307,22 @@ template<> struct Cc_TemplateStruct<I1_Class, I12> { I1_Class* i; };
 // IWYU: I1_Enum is...*badinc-i1.h
 template<class T = I1_Class, I1_Enum E = I11> class Cc_DeclareOnlyTemplateClass;
 
-// I2_Class has a non-explicit constructor (actually, two), so we need
-// the full type here even though it's a const reference.  I1_Class
-// has no implicit, one-argument constructor, so fwd declaring is ok.
 // IWYU: I2_Class needs a declaration
 const I2_Class& Cc_Function(
     // IWYU: I1_Class needs a declaration
     const I1_Class& i1,
-    // IWYU: I2_Class is...*badinc-i2.h.*for autocast
+    // No autocast assurance for function definitions.
```

Comment:
And actually, with the summary suggested above, maybe these `No autocast...` comments are superfluous?

I agree with the other comment cleanups.